### ISSUE TITLE
Error while running locally - Executor failed running

### ISSUE THREAD

**Issue by sankar2389** [sutt0n/serverless-clamav-lambda-layer#2]
=> ERROR [48/52] RUN LD_LIBRARY_PATH=./lib ./bin/freshclam --config-file=bin/freshclam.conf                                                       1.1s 
------
 > [48/52] RUN LD_LIBRARY_PATH=./lib ./bin/freshclam --config-file=bin/freshclam.conf:
#52 0.980 ERROR: Incorrect argument format for option CompressLocalDatabase
#52 0.980 ERROR: Can't open/parse the config file bin/freshclam.conf
------
executor failed running [/bin/sh -c LD_LIBRARY_PATH=./lib ./bin/freshclam --config-file=bin/freshclam.conf]: exit code: 2
Unable to find image 'clamav:latest' locally
docker: Error response from daemon: pull access denied for clamav, repository does not exist or may require 'docker login': denied: requested access to 
the resource is denied.
See 'docker run --help'.
Error: No such container:path: clamav:/home/build/clamav_lambda_layer.zip
Error: No such container: clamav
mv: cannot stat 'clamav_lambda_layer.zip': No such file or directory
~/workspace/serverless-clamav-lambda-layer/layer ~/workspace/serverless-clamav-lambda-layer
unzip:  cannot find or open clamav_lambda_layer.zip, clamav_lambda_layer.zip.zip or clamav_lambda_layer.zip.ZIP.
rm: cannot remove 'clamav_lambda_layer.zip': No such file or directory
~/workspace/serverless-clamav-lambda-layer

OS - Windows 10
I got this error while trying the second step -  ./build.sh

**Comment by sutt0n** [comment]
@sankar2389: Is this the `master` branch, or the `lambda-image` branch?

Edit: The two things I can think of are either try `chmod 777 bin/freshclam.conf` after it's moved into the Docker image during the build process (in the `Dockerfile`) or try changing the line endings since you're using Windows 10 (if that matters?). I'm not near my Windows machine at the moment, but I can see what I can do sometime this week.

**Comment by sankar2389** [comment]
Master branch

**Comment by jpduckwo** [comment]
I had issues on Mac M1 and needed to update the build to specify the architecture with `--platform=linux/x86-64`
This got the lambda-edge branch working for me on Mac with M1. Might not be related but worth a try.
```
#!/bin/bash

rm -rf ./layer
mkdir layer

docker build --platform=linux/x86-64 -t clamav -f Dockerfile .
```

Before that it was failing with

```
 > [layer-image 10/46] RUN yumdownloader -x *i686 --archlist=x86_64 clamav:
#15 0.352 Loaded plugins: ovl, priorities
#15 5.079 64 packages excluded due to repository priority protections
#15 5.154 No Match for argument clamav
#15 5.154 Nothing to download
```

**Comment by RichardAllanBrown** [comment]
I had a similar issue on a Windows 10 machine. I fixed it by ensuring the `freshclam.conf` file used LF not CRLF line endings.

**Comment by sutt0n** [comment]
Apologies for the late reply, y'all.

@jpduckwo Regarding M1/M2 runtime problems, this seems to be the solution. #4 solves this.

**Comment by sutt0n** [comment]
@sankar2389 I pulled the fix from #4 down on my Windows 10 machine and was able to run the build successfully. If you would, confirm whether or not this is still an issue for you or not with the latest version, please and thank you.

**Comment by sutt0n** [comment]
Closing for now.
